tensorflow/python/keras/metrics.py

```
@@ -2856,3 +2856,325 @@ def get(identifier):
+        if average not in (None, 'micro', 'macro', 'weighted'):
+            raise ValueError("Unknown average type. Acceptable values "
+                             "are: [None, micro, macro, weighted]")
+        else:
+            self.average = average
+            if self.average == 'micro':
+                self.axis = None
+            else:
+                self.axis = 0
+        if self.average == 'micro':
+            self.true_positives = self.add_weight(
+                'true_positives',
+                shape=[],
+                initializer='zeros',
+                dtype=self.dtype)
+            self.false_positives = self.add_weight(
+                'false_positives',
+                shape=[],
+                initializer='zeros',
+                dtype=self.dtype)
+            self.false_negatives = self.add_weight(
+                'false_negatives',
+                shape=[],
+                initializer='zeros',
+                dtype=self.dtype)
+        else:
+            self.true_positives = self.add_weight(
+                'true_positives',
+                shape=[self.num_classes],
+                initializer='zeros',
+                dtype=self.dtype)
+            self.false_positives = self.add_weight(
+                'false_positives',
+                shape=[self.num_classes],
+                initializer='zeros',
+                dtype=self.dtype)
+            self.false_negatives = self.add_weight(
+                'false_negatives',
+                shape=[self.num_classes],
+                initializer='zeros',
+                dtype=self.dtype)
+            self.weights_intermediate = self.add_weight(
+                'weights',
+                shape=[self.num_classes],
+                initializer='zeros',
+                dtype=self.dtype)
+
+    # TO DO SSaishruthi: Add sample weight option
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        y_true = tf.cast(y_true, tf.int32)
```

Please raise an exception for unsupported use cases.